### PR TITLE
Allows the use of the content of many2many fields in domains

### PR DESCRIPTION
**Pull request by aricaldeira**
Description of the issue/feature this PR addresses:
Today one cannot use many2many fields in domain/invisible/required view statements, as it's contents are store internally as a array structure

Current behavior before PR:
The following is not possible:

`<field name="many2many_field" />
<field name="some_other_field" attrs="{'invisible': [('many2many_field', '=', False)], 'required': [('many2many_field', '!=', False)]}" />`

Desired behavior after PR is merged:
The example above will be possible, and also the following:

`<field name="some_other_field" attrs="{'invisible': ['!', ('many2many_field', 'child_of', [%(module.xml_id_1)d, %(module.xml_id_2)d, %(module.xml_id_3)d])], 'required': [('many2many_field', 'child_of', [%(module.xml_id_1)d, %(module.xml_id_2)d, %(module.xml_id_3)d])]}" />`

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by renatonlima** [comment]
:+1: It's a nice feature!

@rvalyi @sebastienbeau @alexis-via Can you take a look in this PR?

File: addons/web/static/src/js/framework/data.js [hunk 1]
```diff
@@ -1117,6 +1161,29 @@ function compute_domain (expr, fields) {
...
+                if ((field_value.length == 0) && (val.length == 0)) {
+                    stack.push(true);
+                    break;
+                }
+                if (field_value.length == 0) {
+                    stack.push(false);
+                    break;
+                }
+                if (!_.isArray(val)) val = [val];
+                var itens_esquerda = field_value[0][2];
```
Comment (renatonlima) [inline addons/web/static/src/js/framework/data.js]: Please write variables names in english.

**Comment by tivisse** [comment]
Well, I don't know what to think about this branch. The variables names should be written in English by the way.

@rco-odoo @ged-odoo What do you think about it ?

**Comment by ged-odoo** [comment]
I am not against that kind of changes, but it needs to be done in master, not in 10.0 (this is a stable branch). Please do another PR vs master

**Comment by Yenthe666** [comment]
@aricaldeira and if you do a new PR then please also change the variable names to English ones. 😉

**Pull request closed without merging**